### PR TITLE
Use plenary.strings instead of logic here

### PR DESCRIPTION
**Pull request by delphinus**
This needs https://github.com/nvim-lua/plenary.nvim/pull/96 to be merged.

### CODE REVIEW

File: lua/telescope/utils.lua [hunk 1]
```diff
@@ -249,95 +249,6 @@ function utils.get_os_command_output(cmd, cwd)
   return stdout, ret, stderr
 end
 
-utils.strdisplaywidth = (function()
```
Comment (Conni2461) [inline lua/telescope/utils.lua:252]: Can we not just remove these functions but do something like this
```lua
utils.x = function()
  error("x deprecated. please use plenary.strings.x") 
end
```

**Comment by Conni2461** [comment]
There is more usage of these utils now. (somewhere i can't remember)

File: lua/telescope/utils.lua (hunk 1 above)
Comment (delphinus) [reply lua/telescope/utils.lua:252]: Added on fd1a736

**Comment by Conni2461** [comment]
We have still `utils.strdisplaywidth` in `git.branches` and `lsp.code_action`, which breaks both function in this pr.

**Comment by Conni2461** [comment]
Also you can remove `entry_display.truncate` in telescope afterwards and call `strings.truncate`

**Comment by Conni2461** [comment]
@delphinus I haven't heard from you in quite some time, so i finished it myself. (I hope everything works)

Thanks for all the work on this :)

**Pull request merged**